packages/web-components/src/rating-display/rating-display.ts

```diff
@@ -15,58 +15,107 @@ export class RatingDisplay extends FASTElement {
   public elementInternals: ElementInternals = this.attachInternals();
 
   /**
-   * The color of the rating display icons.
+   * The number of ratings.
    *
    * @public
-   * @default `marigold`
    * @remarks
-   * HTML Attribute: `color`
+   * HTML Attribute: `count`
    */
-  @attr
-  public color?: RatingDisplayColor;
+  @attr({ converter: nullableNumberConverter })
+  public count?: number;
 
   /**
-   * Handles changes to the color attribute.
+   * The maximum possible value of the rating.
+   * This attribute determines the number of icons displayed.
+   * Must be a whole number greater than 1.
    *
-   * @param prev - The previous state
-   * @param next - The next state
+   * @public
+   * @remarks
+   * HTML Attribute: `max`
    */
-  public colorChanged(prev: RatingDisplayColor | undefined, next: RatingDisplayColor | undefined): void {
-    if (prev) toggleState(this.elementInternals, prev, false);
-    if (next) toggleState(this.elementInternals, next, true);
-  }
+  @attr({ converter: nullableNumberConverter })
+  public max?: number;
 
   /**
-   * Renders a single filled icon with a label next to it.
+   * The value of the rating.
    *
    * @public
    * @remarks
-   * HTML Attribute: `compact`
+   * HTML Attribute: `value`
    */
-  @attr({ mode: 'boolean' })
-  public compact: boolean = false;
+  @attr({ converter: nullableNumberConverter })
+  public value?: number;
+
+  private intlNumberFormatter = new Intl.NumberFormat();
+
+  constructor() {
+    super();
+
+    this.elementInternals.role = 'img';
+  }
 
   /**
-   * The number of ratings.
+   * Returns "count" as string, formatted according to the locale.
    *
-   * @public
-   * @remarks
-   * HTML Attribute: `count`
+   * @internal
    */
-  @attr({ converter: nullableNumberConverter })
-  public count?: number;
+  public get formattedCount(): string {
+    return this.count ? this.intlNumberFormatter.format(this.count) : '';
+  }
 
   /**
-   * The maximum possible value of the rating.
-   * This attribute determines the number of icons displayed.
-   * Must be a whole number greater than 1.
+   * Generates the icon SVG elements based on the "max" attribute.
+   *
+   * @internal
+   */
+  public generateIcons(): string {
+    let htmlString: string = '';
+
+    // The value of the selected icon. Based on the "value" attribute, rounded to the nearest half.
+    const selectedValue: number = Math.round((this.value ?? 0) * 2) / 2;
```

Comment:
How about moving this logic to a new method to get the `selectedValue` and calling it here instead, something like `const selectedValue: number = this.getSelectedValue()`? Then in the extended class, we can override that method to account for `this.compact`. That way, we'll have less code duplication.

Reply:
We also have compact referenced int he for loop

Reply:
Right, I'm leaning toward creating two new small methods because it'll be less duplicated code. Or it can also be two new getters for `selectedValue` and `iconCount`.